syntax = "proto3";
package ecommerce.common;

message User {
  string id = 1;
  string name = 2;
  string email = 3;
}

message Product {
  string id = 1;
  string name = 2;
  double price = 3;
  string category = 4;
}

message Order {
  string id = 1;
  string user_id = 2;
  repeated string product_ids = 3;
  double total_price = 4;
  string status = 5;
}

message InventoryItem {
  string product_id = 1;
  int32 quantity = 2;
}

message InventoryStatus {
  string productId = 1;
  bool available = 2;
  int32 quantity = 3;
}

message PaymentResult {
  bool success = 1;
  string transactionId = 2;
  string message = 3;
}

message UpdateInventoryRequest {
  string productId = 1;
  int32 quantityChange = 2; // Negative to reduce stock
}
